### ISSUE TITLE
Form returns whole page instead of message

### ISSUE THREAD

**Issue by johannesE** [getgrav/grav-plugin-form#195]
I have an ajax form (basically followed the tutorial).
`pages/forms/default.de.md`:
```title: Forms
forms:
    footer-contact-form:
        template: form-messages
        refresh_prevention: false
        fields:
            name:
                label: Name
                type: text
                validate: { required: true }
            email:
                label: Email
                type: email
                placeholder: 'z. B. example@octotreat.ch'
            phone:
                label: Telefonnummer
                type: text
                placeholder: 'z. B. +41 41 508 58 43'
            message:
                label: Nachricht
                type: textarea
                validate: { required: true }
        buttons:
            submit:
                type: submit
                value: Abschicken
        process:
            message: 'Danke für Deine Nachricht. Wir kommen so schnell wie möglich auf Dich zurück.'
            reset: true
            email:
                from: '{{ config.plugins.email.from }}'
                to: '{{ config.plugins.email.to }}'
                subject: 'Contact by {{ form.value.name|e }}'
                body: '{% include ''forms/data.html.twig'' %}'
```
Please note that I include the`form-messages` template.

My javascript code (without jquery, but the outcome with jquery is the same). 
```
document.addEventListener('DOMContentLoaded', function() {
    let form = document.getElementById('footer-contact-form');
    form.onsubmit = function(e) {
      e.preventDefault();

      var request = new XMLHttpRequest();
      request.open(form.getAttribute('method'), form.getAttribute('action'), true);
      request.setRequestHeader('Content-Type', 'application/html; charset=UTF-8');
      request.setRequestHeader('Accept', 'application/html; charset=UTF-8');

      request.onload = function () {
        if (request.status >= 200 && request.status < 400) {
          // Success!
          var resp = request.responseText;
          document.getElementById('footer-form-result').innerHTML = resp;
          console.log(request.status);
        } else {
          // We reached our target server, but it returned an error
          // TODO
          console.log("error")
        }
      };

      request.onerror = function () {
        // There was a connection error of some sort
        // TODO
      };

      request.send(new FormData(form));
    };
});
```
And my footer: 

        <div class='form'>
            <div id="footer-form-result"></div>
            {% include "forms/form.html.twig" with { form: forms('footer-contact-form') } %}
        </div>

The server sends me the whole current page as the response, which is then packed into the `#footer-form-result`. The response has the 200 status.

I updated the form plugin to the newest version (v2.9.2). I even downloaded the repo as zip and tried that. Same output.

The theme has been created with `$ bin/plugin devtools new-theme`, if that helps, but I suspect the problem is elsewhere.

**Comment by rhukster** [comment]
This technique does send back the HTML pre-rendered as part of the form-messages template, the only thing is the form does not send back an error code if there is an error, it simply sends back the messages with 200 response even if there is an error message in there.

The main reason is that we don't actually know when the response comes back if it was an error or not, or if it contains 1 error and 1 success, etc.  Its just a list of messages.  Short of iterating over the messages and setting the response to something other than 200 if an error message is found, i'm not sure the best way to handle this.

**Comment by rhukster** [comment]
Can we continue the discussion here: https://github.com/getgrav/grav-plugin-form/issues/163#issuecomment-328243387

As it sounds like the same issue.

**Comment by johannesE** [comment]
@rhukster #163 should be fixed, no?. My problem is not, that I'm not receiving the correct HTTP code, but rather that I'm receiving no (error or success) message at all.

What I should be receiving:
```<div class="alert notices {{ form.message_color ?: 'green' }}"><p>{{ message|raw }}</p></div>```
Where the message is "Danke für Deine Nachricht. Wir kommen so schnell wie möglich auf Dich zurück.".

What I actually receive:
```<meta charset="utf-8">
<title>Home | Octotreat - Human focused applications</title>
[...]
<main>
[...]
<div id="footer-form-result">[nothing in here]</div>
[here comes the empty form. no messages. The fields are also empty, which is probably due to the reset: true option.]
```

**Comment by openscript** [comment]
If a form lives on the document root page `/` and i18n is activated, it's not going to work.

Let's imagine we are on the root page in the language German so the path is `example.com/de`. If there is a form rendered, without an action path defined, the action would be:

![image](https://user-images.githubusercontent.com/1105080/31122963-4723bf06-a83e-11e7-92e3-5541d83055f0.png)

So if the form is submitted a POST request is sent to `/de/`. The page at `/de/` redirects to `/de`. During that process the POST data is lost and the whole page at `/de` is returned.

Probably the way the action is defined by default is wrong. `page.route` adds a slash to the end of the action.
![image](https://user-images.githubusercontent.com/1105080/31122947-36eac77e-a83e-11e7-8001-0f376c4090b0.png)

I think a fix for this bug should be to remove any trailing slashes from the action. Should I create a PR?

**Comment by johannesE** [comment]
@openscript good find!
I tried to set my form action accordingly to `/de`, this avoids the redirect but the server still returns the whole page.

**Comment by rhukster** [comment]
Pr would b great :) @openscript

**Comment by mgollmer** [comment]
Hello there, 
I'm facing the same Problem :(. I have the latest plugin version and there is still the whole page coming through. 

I solved it as explained in https://learn.getgrav.org/forms/forms/how-to-ajax-submission.

Regards,
Michael

**Comment by mgollmer** [comment]
Somebody found a solution? I need to use this kind form in my project =(

**Comment by johannesE** [comment]
The only solution we found is not to use ajax. We also defined the action in the form, but I think this is not necessary anymore because of the PR from @openscript.

**Comment by rhukster** [comment]
Pretty sure this is all working now: https://github.com/getgrav/grav-plugin-form/issues/163

**Comment by parijke** [comment]
Hi @rhukster 

I am using the latest version 2.10.0 but it stil get the whole html page returned, even when I use form-messages as the template

**Comment by mgollmer** [comment]
Hello there, 

can you provide your frontmatter for your form? I had the same error and solved it. My YAML-Code for the form looks like this.

```

form:
    name: contact-form
    id: contact
    template: form-messages
    classes: 'g-grid cs_contact_cont-form'
    fields:
.
.
.
 process:
        -
            email:
                from: '{{ config.plugins.email.from }}'
                to: ['{{ config.plugins.email.to }}']
                subject: '[Feedback] {{ form.value.name|e }}'
                body: '{% include ''forms/data.html.twig'' %}'
        -
            save:
                fileprefix: feedback-
                dateformat: Ymd-His-u
                extension: txt
                body: '{% include ''forms/data.txt.twig'' %}'
        -
            message: blabla
        -
            reset: true
```

**Comment by parijke** [comment]
here's mine
`---
title: 'Offerte aanvraag'
slug: offerte-aanvraag
debugger: true
cache_enable: false
routable: false
visible: false
refresh_prevention: true
form:
    name: offerte-aanvraag
    action: /offer/ajax-process
    template: form-messages
    classes: form-classes
    fields:
        -
            name: name
            outerclasses: outerclasses
            classes: classes
            wrapper_classes: wrapper_classes
            #style: 'color:blue;'
            labelclasses: labelclasses
            dataclasses: dataclasses
            label: Naam
            placeholder: 'Uw volledige naam'
            autofocus: 'on'
            autocomplete: 'on'
            type: text
            validate:
                required: true

        -
            name: info
            label: 'Uw wilt info over'
            type: checkboxes
            options:
                Zorgverzekering: Zorgverzekering
                Woonverzekering: Woonverzekering
                Autoverzekering: Autoverzekering
                Hypotheken: Hypotheken
                Aankoopadvies: Aankoopadvies
                Verhuur: Verhuur
                Beheer_onderhoud: Beheer en onderhoud
        -
            name: provincie
            label: Provincie
            autofocus: 'on'
            autocomplete: 'on'
            type: select
            options:
                Alicante: Alicante
                Almeria: Almeria
                Asturias: Asturias
                Ávila: Ávila
                Badajoz: Badajoz
                Baleares: Baleares
                Barcelona: Barcelona
                Burgos: Burgos
                Caceres: Caceres
                Cádiz: Cádiz
                Cantabria: Cantabria
                Castellón: Castellón
                Ceuta: Ceuta
                Ciudad Real: Ciudad Real
                Cordoba: Cordoba
                Coruña: Coruña
                Cuenca: Cuenca
                Gerona: Gerona
                Granada: Granada
                Guadalajara: Guadalajara
                Guipúzcoa: Guipúzcoa
                Huelva: Huelva
                Huesca: Huesca
                Jaén: Jaén
                Las Palmas: Las Palmas
                León: León
                Lerida: Lerida
                Lugo: Lugo
                Madrid: Madrid
                Malaga: Malaga
                Melilla: Melilla
                Murcia: Murcia
                Navarra: Navarra
                Orense: Orense
                Plencia: Plencia
                Pontevedra: Pontevedra
                Rioja: Rioja
                Salamanca: Salamanca
                Segovia: Segovia
                Sevilla: Sevilla
                Soria: Soria
                Tarragona: Tarragona
                Tenerife: Tenerife
                Teruel: Teruel
                Toledo: Toledo
                Valencia: Valencia
                Vizcaya: Vizcaya
                Zamora: Zamora
                Zaragoza: Zaragoza
        -
            name: telefoon
            label: Telefoon
            placeholder: 'Uw telefoonnummer'
            type: tel
            validate:
                required: true
        -
            name: email
            label: Email
            placeholder: 'Enter your email address'
            type: email
            validate:
                required: true
    buttons:
        -
            type: submit
            classes: btn klasse
            value: Verzend

    process:
        -
            email:
                from: '{{ config.plugins.email.from }}'
                to: ['{{ config.plugins.email.to }}', '{{ form.value.email }}']
                subject: '[Aanvraag {{ form.value.type|lower|e }} voor {{ form.value.name|e}}]'
                body: '{% include ''forms/data.html.twig'' %}'
        -
            save:
                fileprefix: feedback-
                dateformat: Ymd-His-u
                extension: txt
                body: '{% include ''forms/data.txt.twig'' %}'
        -
            message: 'Dank voor uw aanvraag!'
        -
            display: /offer/thankyou
---`

**Comment by mgollmer** [comment]
Dele there "action" and "display" parameter and try again.

**Comment by parijke** [comment]
ahhhhhhhhh it works

**Comment by parijke** [comment]
thank you so much @mgollmer

**Comment by mgollmer** [comment]
you're welcome @parijke
